### ISSUE TITLE
Upgrading from 4.5.1 to 4.6.0

### ISSUE THREAD

**Issue by ssiahetiong** [CrunchyData/postgres-operator#2239]
**CrashLoopBackOff for database pod**

```
Fri Jan 29 14:13:57 UTC 2021 INFO: Applying SSHD..
Fri Jan 29 14:13:57 UTC 2021 INFO: Checking for SSH Host Keys in /sshd..
Fri Jan 29 14:13:57 UTC 2021 INFO: Checking for authorized_keys in /sshd
Fri Jan 29 14:13:57 UTC 2021 INFO: Checking for sshd_config in /sshd
Fri Jan 29 14:13:58 UTC 2021 INFO: setting up .ssh directory
Fri Jan 29 14:13:58 UTC 2021 INFO: Starting SSHD..
Fri Jan 29 14:13:58 UTC 2021 INFO: Starting background process to monitor Patroni initization and restart the database if needed
Fri Jan 29 14:13:58 UTC 2021 INFO: Now removing "pause" key from patroni.dynamic.json configuration file if present
Fri Jan 29 14:13:58 UTC 2021 INFO: Initializing cluster bootstrap with command: '/usr/local/bin/patroni /tmp/postgres-ha-bootstrap.yaml'
Fri Jan 29 14:13:58 UTC 2021 INFO: Running Patroni as PID 1
2021-01-29 14:01:05,320 INFO: No PostgreSQL configuration items changed, nothing to reload.
2021-01-29 14:01:05,321 ERROR: [Errno 13] Permission denied: '/crunchyadm'
Process Process-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/patroni/utils.py", line 461, in validate_directory
    os.makedirs(d)
  File "/usr/lib64/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/crunchyadm'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 138, in patroni_main
    abstract_main(Patroni, schema)
  File "/usr/local/lib/python3.6/site-packages/patroni/daemon.py", line 98, in abstract_main
    controller = cls(config)
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 33, in __init__
    self.postgresql = Postgresql(self.config['postgresql'])
  File "/usr/local/lib/python3.6/site-packages/patroni/postgresql/__init__.py", line 76, in __init__
    self.config.check_directories()
  File "/usr/local/lib/python3.6/site-packages/patroni/postgresql/config.py", line 335, in check_directories
    self.try_to_create_dir(d.strip(), "'{}' is defined in unix_socket_directories, {}")
  File "/usr/local/lib/python3.6/site-packages/patroni/postgresql/config.py", line 330, in try_to_create_dir
    validate_directory(d, msg)
  File "/usr/local/lib/python3.6/site-packages/patroni/utils.py", line 464, in validate_directory
    raise PatroniException(msg.format(d, "couldn't create the directory"))
patroni.exceptions.PatroniException: "'/crunchyadm' is defined in unix_socket_directories, couldn't create the directory"
```

 
 - [x] bug report
 - [ ] feature request
 - [ ] support request

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce:**
Upgraded postgres operator to 4.6.0 successfully. When I performed a pgo upgrade command, the database pod is crashing.

**What is the expected behavior?**

**What is the motivation or use case for changing the behavior?**

**Other information** (e.g. detailed explanation, related issues, suggestions how to fix, etc)

**Please tell us about your environment:**

 - Operating System: Centos 7
 - Container Image Tag: centos8-13.1-4.6.0
 - Operator Version: 4.6.0
 - Storage (NFS, hostpath, storage class): hostpath
 - PostgreSQL Version: 13.1
 - Platform (Docker, Kubernetes, OpenShift): kubernetes
 - Platform Version: 1.20.1

**Comment by jkatz** [comment]
How did you upgrade?

**Comment by ssiahetiong** [comment]
I have an existing 4.5.1 cluster, then I used pgo upgrade "cluster" command to update the cluster version. Somehow I was able to figure it out. The previous version of (cluster)-pgha-config have this line:
 unix_socket_directories: /tmp,/crunchyadm

I modified it to unix_socket_directories: /tmp and the error was gone. I also found out that the '/opt/cpm' path have changed to /opt/crunchy and I have to update the configmap setting.

**Comment by jkatz** [comment]
@ssiahetiong Thanks, I wanted to first ensure you were using the standard path.

I think we may need to add something to the upgrade script to remove the `/crunchyadm` UNIX socket directory. I'll reclassify this as a bug so we can work on something for 4.6.1.

**Comment by jkatz** [comment]
To keep items around upgrading consolidated, #2250 fixes an issue where a surprise `pgBadger` sidecar could be deployed with an upgraded Postgres instance.

If you encounter this and you do not want a pgBadger sidecar, the fix is simple and can be handled in one of two ways:

1. `pgo update cluster --disable-pgbadger $CLUSTER_NAME`.
2. Update the `pgclusters.crunchydata.com` custom resource and set `pgBadger` to `false`.

**Comment by tchellomello** [comment]
@jkatz  hitting the same problem, however, after the `pgo update cluster --disable-pgbadger $CLUSTER_NAME` still not there. Any ideas?

```
$ pgo version
pgo client version 4.6.0
pgo-apiserver version 4.6.0

$ pgo show cluster toca

cluster : toca (crunchy-postgres-ha:centos8-13.1-4.6.0)
	pod : toca-tgcy-cdf858cbf-vbvkm (Running) on r610n1 (1/2) (unknown)
		pvc: toca-tgcy (20Gi)
	resources : Memory: 128Mi
	deployment : toca-backrest-shared-repo
	deployment : toca-tgcy
	service : toca - ClusterIP (10.233.39.128) - Ports (9187/TCP, 2022/TCP, 5432/TCP)
	service : toca-pgbouncer - ClusterIP (10.233.57.166) - Ports (5432/TCP)
	service : toca-replica - ClusterIP (10.233.20.200) - Ports (9187/TCP, 2022/TCP, 5432/TCP)
	pgreplica : toca-isqx
	labels : crunchy-postgres-exporter=true pgo-backrest=true pgo-version=4.6.0 pgouser=admin workflowid=8e8ca592-eb4f-4813-9610-0e051a3aaf54 crunchy-pgha-scope=toca deployment-name=toca-tgcy name=toca pg-cluster=toca pg-pod-anti-affinity=preferred 

$ pgo update cluster --disable-pgbadger toca
Adding or removing a pgBadger sidecar can cause downtime.
WARNING: Are you sure? (yes/no): yes
updated pgcluster toca

** I don't see pgBagder**
$ kubectl get crd pgclusters.crunchydata.com -o yaml  | grep -i bad
                  f:pgbadgerport:
              pgbadgerport:

$ pgo update cluster toca --shutdown
WARNING: Are you sure? (yes/no): yes
updated pgcluster toca

kubectl get pods -n pgo -w -o wide                                                                                                          15:33:33
NAME                                READY   STATUS      RESTARTS   AGE   IP              NODE     NOMINATED NODE   READINESS GATES
pgo-deploy-sc59q                    0/1     Completed   0          35m   10.233.68.145   r610n2   <none>           <none>
postgres-operator-797bcb5d6-p7j76   4/4     Running     1          33m   10.233.68.147   r610n2   <none>           <none>

$ pgo upgrade toca 
All active replicas will be scaled down and the primary database in this cluster will be stopped and recreated as part of this workflow!
WARNING: Are you sure? (yes/no): yes
created upgrade task for toca
```

Then I see this:

```
 kubectl get pods -n pgo -o wide                                                                                                             15:36:28
NAME                                         READY   STATUS             RESTARTS   AGE   IP              NODE     NOMINATED NODE   READINESS GATES
pgo-deploy-sc59q                             0/1     Completed          0          37m   10.233.68.145   r610n2   <none>           <none>
postgres-operator-797bcb5d6-p7j76            4/4     Running            1          36m   10.233.68.147   r610n2   <none>           <none>
toca-backrest-shared-repo-86bf6fc54d-262nw   1/1     Running            0          17s   10.233.68.195   r610n2   <none>           <none>
toca-tgcy-686569b4b6-tvkwq                   1/2     CrashLoopBackOff   1          15s   10.233.67.21    r610n1   <none>           <none>

[toca-tgcy-686569b4b6-tvkwq database] Sat Feb 13 20:36:22 UTC 2021 INFO: setting up .ssh directory 
[toca-tgcy-686569b4b6-tvkwq database] Sat Feb 13 20:36:22 UTC 2021 INFO: Starting SSHD.. 
[toca-tgcy-686569b4b6-tvkwq database] Sat Feb 13 20:36:22 UTC 2021 INFO: Starting background process to monitor Patroni initization and restart the database if needed 
[toca-tgcy-686569b4b6-tvkwq database] Sat Feb 13 20:36:22 UTC 2021 INFO: Now removing "pause" key from patroni.dynamic.json configuration file if present 
[toca-tgcy-686569b4b6-tvkwq database] Sat Feb 13 20:36:22 UTC 2021 INFO: Initializing cluster bootstrap with command: '/usr/local/bin/patroni /tmp/postgres-ha-bootstrap.yaml' 
[toca-tgcy-686569b4b6-tvkwq database] Sat Feb 13 20:36:22 UTC 2021 INFO: Running Patroni as PID 1 
[toca-tgcy-686569b4b6-tvkwq database] 2021-02-13 20:36:23,267 INFO: No PostgreSQL configuration items changed, nothing to reload. 
[toca-tgcy-686569b4b6-tvkwq database] 2021-02-13 20:36:23,268 ERROR: [Errno 13] Permission denied: '/crunchyadm' 
[toca-tgcy-686569b4b6-tvkwq database] Process Process-1: 
[toca-tgcy-686569b4b6-tvkwq database] Traceback (most recent call last): 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/utils.py", line 461, in validate_directory 
[toca-tgcy-686569b4b6-tvkwq database]     os.makedirs(d) 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/lib64/python3.6/os.py", line 220, in makedirs 
[toca-tgcy-686569b4b6-tvkwq database]     mkdir(name, mode) 
[toca-tgcy-686569b4b6-tvkwq database] PermissionError: [Errno 13] Permission denied: '/crunchyadm' 
[toca-tgcy-686569b4b6-tvkwq database]  
[toca-tgcy-686569b4b6-tvkwq database] During handling of the above exception, another exception occurred: 
[toca-tgcy-686569b4b6-tvkwq database]  
[toca-tgcy-686569b4b6-tvkwq database] Traceback (most recent call last): 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap 
[toca-tgcy-686569b4b6-tvkwq database]     self.run() 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run 
[toca-tgcy-686569b4b6-tvkwq database]     self._target(*self._args, **self._kwargs) 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 138, in patroni_main 
[toca-tgcy-686569b4b6-tvkwq database]     abstract_main(Patroni, schema) 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/daemon.py", line 98, in abstract_main 
[toca-tgcy-686569b4b6-tvkwq database]     controller = cls(config) 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 33, in __init__ 
[toca-tgcy-686569b4b6-tvkwq database]     self.postgresql = Postgresql(self.config['postgresql']) 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/postgresql/__init__.py", line 76, in __init__ 
[toca-tgcy-686569b4b6-tvkwq database]     self.config.check_directories() 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/postgresql/config.py", line 335, in check_directories 
[toca-tgcy-686569b4b6-tvkwq database]     self.try_to_create_dir(d.strip(), "'{}' is defined in unix_socket_directories, {}") 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/postgresql/config.py", line 330, in try_to_create_dir 
[toca-tgcy-686569b4b6-tvkwq database]     validate_directory(d, msg) 
[toca-tgcy-686569b4b6-tvkwq database]   File "/usr/local/lib/python3.6/site-packages/patroni/utils.py", line 464, in validate_directory 
[toca-tgcy-686569b4b6-tvkwq database]     raise PatroniException(msg.format(d, "couldn't create the directory")) 
[toca-tgcy-686569b4b6-tvkwq database] patroni.exceptions.PatroniException: "'/crunchyadm' is defined in unix_socket_directories, couldn't create the directory" 
```

**Comment by tchellomello** [comment]
I was able to modify the (cluster)-pgha-config however, now I'm hitting #2094

**Comment by jkatz** [comment]
This issue is fixed in #2242 as this thread mentions, and will be released in 4.6.1.
